Homework week 3 <Reshad Majidi>

majidireshad opened the pull request:
Finished week 3 homework exercises and tested for errors.

J05J4 commented on 1-JavaScript/Week3/homework/ex7-mindPrivacy.js:
```diff
@@ -28,8 +28,14 @@ const employeeRecords = [
-  // TODO complete this function
+function filterPrivateData(filteredRecords) {
+  const array=[];
+for (const obj of filteredRecords){
```
```suggestion
for (const employeeRecord of filteredRecords){
```
`obj` does not say anything about the object you are working with.

J05J4 commented on 1-JavaScript/Week3/homework/ex7-mindPrivacy.js:
```diff
@@ -28,8 +28,14 @@ const employeeRecords = [
+function filterPrivateData(filteredRecords) {
+  const array=[];
+for (const obj of filteredRecords){
+   const {name,occupation,email}=obj;
+    const newObj={name,occupation,email};
+  array.push(newObj);
```
You could simplify this a bit by doing this:
```suggestion
   const {name, occupation, email} = employeeRecord;
   array.push({name,occupation,email});
```

And you can even do the reversed thing, that is even more advanced: 
```
   const {gender, salary, ...publicData} = employeeRecord;
   array.push(publicData);
```

J05J4 commented on 1-JavaScript/Week3/homework/ex7-mindPrivacy.js:
```diff
@@ -28,8 +28,14 @@ const employeeRecords = [
-function filterPrivateData(/* parameter(s) go here */) {
-  // TODO complete this function
+function filterPrivateData(filteredRecords) {
+  const array=[];
```
```suggestion
  const filteredEmployeeRecords = [];
```

J05J4 reviewed (changes requested):
This is a good implementation of the homework. You seem to understand most of the topics in the curriculum. 
However you should take more care on the formatting of the code: I've made a lot of comments on those, don't be scared off the number of comments; most are with a suggestion, try to check if you understand the suggestion, and implement them in all the code that you have written this week. 
You should also take better care on naming arguments and variables: an array can never have the name `array` that is not descriptive. Someone who is reading your code should understand by the name what it is you're working with. This will also help you when you are writing code, because it is forcing you to think about the type you expect. 
In `ex5` you have introduced recursion in your solution: I do not think this was intended and this implementation is not correct: you should fix that part. 
Other than that you did a good job and you solved all the exercises!

majidireshad commented on 1-JavaScript/Week3/homework/ex6-totalCost.js:
```diff
@@ -19,26 +20,38 @@ instead!
+function calculateTotalPrice(total) {
+ 
+  const objectValues=Object.values(total);
+  const reducer = (a, b) => a + b;
+  const sumCart=objectValues.reduce(reducer);
```
I actually know more about reduce by this week! I can use a simple for loop to do this: 
```
let sumCart = 0;
for (let amount of objectValues){
sumCart +=  amount;
}
```

majidireshad commented:
Hi @J05J4, Thank you a lot for the detailed feedback. I updated the files. I simplified some of the codes as well as assigned more appropriate names to the variables. I fixed the formatting and understood the difference it makes. Furthermore, I corrected the implementation mistakes and tested the files.
